### PR TITLE
feat: #153

### PR DESCRIPTION
**Pull request by hugomrdias**
- typedoc support to generate docs for all the clients
- add list of Abilities types
- renamed `any.js` to `wildcard.js`
- added `parseAbility` and `canDelegateAbility` functions to the capabilities `utils.js`
- cli is mostly commented out needs to be updated in a follow PR
- StoreData interface revamped
- added more exports to index.js to help with the generated docs until typedoc supports export maps
- added a bunch of delegation utilities to `delegation.js`
- agent has new methods, `addProof`, `proofs`, `delegations`, `delegationsWithMeta`, `createSpace`, `setCurrentSpace`, `currentSpace`, `currentSpaceWithMeta` and `execute`
- old createAccount renamed to `registerSpace`
- a bunch of jsdocs 

preview docs https://vigorous-chandrasekhar-6b427f.netlify.app/

todo:

- [x] rename caps
- [x] rename account to space in api
- [x] finish setup wrangler migrations for staging and prod (using CI)
- [x] more docs for agent methods 
- [x] feedback
- [x] support multiple invocations to be executed on a connection

### CODE REVIEW

File: packages/access-client/src/agent.js [hunk 1]
```diff
@@ -128,67 +126,240 @@ export class Agent {
...
+    }
+
+    await this.store.save(this.data)
+  }
+
+  /**
+   * @param {import('@ucanto/interface').Capability[]} [caps]
+   */
+  async *proofs(caps) {
+    for await (const value of this.proofsWithMeta(caps)) {
```
Comment (alanshaw) [inline packages/access-client/src/agent.js:180]: Can they please be collected for me? I'm not sure when I would not just collect them all before sending them...

File: packages/access-client/src/cli/cmd-whoami.js [hunk 2]
```diff
@@ -1,33 +1,31 @@
...
+  //   console.log('Delegations received:')
+  //   for (const [key, value] of delegations.receivedByResource) {
+  //     console.log(
+  //       `Resource: ${key}`,
+  //       value.map((cap) => cap.cap.can)
+  //     )
+  //   }
+  // } else {
+  //   console.error(`Run "${NAME} setup" first`)
+  // }
```
Comment (alanshaw) [inline packages/access-client/src/cli/cmd-whoami.js:30]: Needs implementation?

File: packages/access-client/src/cli/index.js [hunk 3]
```diff
@@ -81,111 +39,103 @@ prog
...
+    //   try {
+    //     const result = await agent.getAccountInfo(account)
+    //     console.log(result)
+    //   } catch (error_) {
+    //     const error = /** @type {Error} */ (error_)
+    //     console.log(error.message)
+    //   }
+    // } else {
+    //   console.error(`Run "${NAME} setup" first`)
+    // }
```
Comment (alanshaw) [inline packages/access-client/src/cli/index.js:75]: Needs implementation?

File: packages/access-client/src/stores/store-indexeddb.js [hunk 4]
```diff
@@ -41,6 +40,10 @@ export class StoreIndexedDB {
     this.#dbStoreName = options.dbStoreName ?? STORE_NAME
   }
 
+  /**
+   *
+   * @returns {Promise<Store>}
```
Comment (alanshaw) [inline packages/access-client/src/stores/store-indexeddb.js:45]: If you declare that these instance methods return a `Store` then you lose the ability to use it in specifics. It should adhere to that interface _anyway_, you shouldn't need to explicitly state it.

There's a bunch of these that have been added - do we really need to?

File: packages/access-client/test/awake.node.test.js [hunk 5]
```diff
@@ -41,12 +40,12 @@ describe('awake', function () {
   it('should send msgs', async function () {
     const agent1 = await Agent.create({
       store: await StoreMemory.create(),
-      fetch: globalThis.fetch || fetch,
       url: new URL('http://127.0.0.1:8787'),
     })
+    const account = await agent1.createSpace('responder')
```
Comment (alanshaw) [inline packages/access-client/test/awake.node.test.js:45]: Still referring to "account" here?

File: packages/access-client/src/cli/cmd-whoami.js (hunk 2 above)
Comment (hugomrdias) [reply packages/access-client/src/cli/cmd-whoami.js:30]: im gonna rework the cli in a follow up

File: packages/access-client/src/cli/index.js (hunk 3 above)
Comment (hugomrdias) [reply packages/access-client/src/cli/index.js:75]: im gonna rework the cli in a follow up

File: packages/access-client/src/stores/store-indexeddb.js (hunk 4 above)
Comment (hugomrdias) [reply packages/access-client/src/stores/store-indexeddb.js:45]: This `Store` type is the local one defined at the top of the file, its already specific. If you dont do this you loose the method documentation defined in the types.ts file for the interface.

What are we losing here ?

Comment (alanshaw) [reply packages/access-client/src/stores/store-indexeddb.js:45]: Oh ok I thought this was referring to the interface!

**Review by alanshaw — APPROVED** [review]

**Pull request merged**